mos_tests/functions/common.py

```diff
@@ -229,183 +231,266 @@ def download_image(image_link_file, where_to_put='/tmp/'):
+                       if s.id == uid][0]
         while inst_status != status and time() < start_time + 60 * timeout:
             sleep(1)
-            inst_status = [s.status for s in cinderclient.volumes.list() if
-                           s.id == uid][0]
-        if inst_status == status:
-            return True
+            inst_status = [s.status for s in cinder_client.volumes.list()
+                           if s.id == uid][0]
+        return inst_status == status
     return False
+
+
+# Flavor functions
+def is_flavor_exists(nova_client, flavor_id):
+    """ Check the presence of flavor in the system
+        :param nova_client: Nova API client connection point
+        :param flavor_id: name of the flavor
+        :return True or False
+    """
+    return flavor_id in [f.id for f in nova_client.flavors.list()]
+
+
+def get_flavor_id_by_name(nova_client, flavor_name):
+    """ The function returns flavor's id by its name
+        :param nova_client: Nova API client connection point
+        :param flavor_name: Name of the flavor
+        :return: UID of the flavor
+        None if the flavor with required name does not exist
+    """
+    for flavor in nova_client.flavors.list():
+        if flavor.name == flavor_name:
+            return flavor.id
+    return None
+
+
+def delete_flavor(nova_client, flavor_id):
+    """ This function delete the flavor by its name.
+        :param nova_client: Nova API client connection point
+        :param flavor_id: UID of the flavor to delete
+        :return: Nothing
+    """
+    for flavor in nova_client.flavors.list():
+        if flavor.id == flavor_id:
+            nova_client.flavors.delete(flavor)
+            break
+    while is_flavor_exists(nova_client, flavor_id):
+        sleep(1)
+
+
+# Images
+def is_image_exists(glance_client, image_id):
+    """ This function check if image with required id presents in the system
+        or not
+        :param glance_client: Glance API client connection point
+        :param image_id: UID of the image to delete
+        :return: True if the image with provided id presents in the system;
+        False otherwise
+    """
+    return image_id in [image.id for image in glance_client.images.list()]
+
+
+def delete_image(glance_client, image_id):
+    """ This function should delete the image with provided id from the system
+        :param glance_client: Glance API client connection point
+        :param image_id: UID of the image to delete
+        :return: Nothing
+    """
+    glance_client.images.delete(image_id)
+    while is_image_exists(glance_client, image_id):
+        sleep(1)
+
+
+# execution of system commands
+def ping_command(ip_address):
+    """ This function executes the ping program and check its results
+        :param ip_address: The IP address to ping
+        :return: True in case of success, False otherwise
+    """
+    the_count_key = '-c'
+    if platform.system() == 'Windows':
```

Comment:
So, we shouldn't plan compatibility with Windows, right? For any future activities?